panel/src/panel/content.js

```diff
@@ -1,130 +1,151 @@
+		get hasChanges() {
+			return length(this.changes) > 0;
 		},
+
 		/**
-		 * Whether the model is a draft
+		 * Whether content is currently being discarded, saved or published
 		 *
 		 * @returns {Boolean}
 		 */
-		get isDraft() {
-			return panel.view.props.model.status === "draft";
-		},
-		isPublishing: false,
-		isSaving: false,
+		isProcessing: false,
+
 		/**
-		 * Whether the content is currently locked by another user
+		 * The last published state
 		 *
-		 * @returns {Boolean}
+		 * @returns {Object}
 		 */
-		get isLocked() {
-			return this.lock?.state === "lock";
+		get originals() {
+			return panel.view.props.originals;
 		},
+
 		/**
-		 * Content lock state of the model
-		 *
-		 * @returns {Object|null|false}
+		 * Publishes any changes
 		 */
-		get lock() {
-			const lock = panel.view.props.lock;
-
-			if (!lock) {
-				return false;
+		async publish() {
+			if (this.isProcessing === true) {
+				return;
 			}
 
-			if (lock.state === null) {
-				return null;
+			this.isProcessing = true;
+
+			// Send updated values to API
+			try {
+				await panel.post(
+					this.api + "/changes/publish",
+					panel.view.props.content
+				);
+
+				panel.view.props.originals = panel.view.props.content;
+
+				await panel.view.refresh();
+			} finally {
+				this.isProcessing = false;
 			}
 
-			return {
-				...lock.data,
-				state: lock.state
-			};
-		},
-		/**
-		 * Publishes any changes
-		 */
-		async publish() {
-			this.isPublishing = true;
-			await panel.app.$store.dispatch("content/save");
 			panel.events.emit("model.update");
 			panel.notification.success();
-			this.isPublishing = false;
-		},
-		/**
-		 * Returns all fields and their already published values
-		 *
-		 * @returns {Object}
-		 */
-		get published() {
-			return panel.app.$store.getters["content/originals"]();
 		},
+
 		/**
 		 * Saves any changes
 		 */
 		async save() {
-			this.isSaving = true;
-			// …
-			this.isSaving = false;
+			if (this.isProcessing === true) {
+				return true;
+			}
+
+			this.isProcessing = true;
+
+			try {
+				await panel.post(this.api + "/changes/save", panel.view.props.content);
+			} finally {
+				this.isProcessing = false;
+			}
 		},
+
 		/**
 		 * Updates the values of fields
 		 *
-		 * @param {Object} fields
-		 * @param {any} value
+		 * @param {Object} values
 		 */
-		set(fields) {
-			panel.app.$store.dispatch("content/update", [null, fields]);
+		update(values) {
+			if (length(values) === 0) {
+				return;
+			}
+
+			panel.view.props.content = {
+				...panel.view.props.originals,
+				...values
+			};
 		},
-		/**
-		 * Removes the content lock for the current user,
-		 * e.g. when closing/leaving the model view
-		 */
-		async unlock() {},
+
 		/**
 		 * Returns all fields and values incl. changes
 		 *
 		 * @returns {Object}
 		 */
 		get values() {
-			return {
-				...this.published,
-				...this.changes
-			};
+			return panel.view.props.content;
```

Comment:
How about adding also 

```
set values(values) {
  panel.view.props.content = values
}
```

So that we can stay more inside the module logic with less references directly to view props?

Reply:
Same as above. If we test it and it properly works, I'm fine with it. But I had issues with getters and setters before.

Reply:
I get what you mean, let's keep it, not an important change request.